### ISSUE TITLE
"panic: interface conversion" while trying to shape

### ISSUE THREAD

**Issue by philrz** [brimsec/zq#2219]
Repro is with `zq` commit `1ac7787f`.

I was trying to verify #2198 and was hacking around before fully understanding the example, and triggered a panic.

```
$ cat example.zs 
type port=uint16;
type conn={orig_h:ip,orig_p:port,resp_h:ip,resp_p:port};
put . = shape(., {_path:string,ts:time,uid:bstring,id:conn})

$ echo '{"foo": "bar"}' | zq -z -I example.zs -
panic: interface conversion: zng.Type is *zng.TypeAlias, not *zng.TypeRecord
goroutine 68 [running]:
runtime/debug.Stack(0x46, 0x40, 0x6b)
	/usr/local/opt/go/libexec/src/runtime/debug/stack.go:24 +0x9f
github.com/brimsec/zq/zqe.RecoverError(0x182e8c0, 0xc0003bc690, 0xc0003bc690, 0xc000616200)
	/Users/phil/work/zq/zqe/error.go:163 +0x26
github.com/brimsec/zq/driver.(*mux).safeGet.func1(0xc000625f70)
	/Users/phil/work/zq/driver/mux.go:49 +0x48
panic(0x182e8c0, 0xc0003bc690)
	/usr/local/opt/go/libexec/src/runtime/panic.go:969 +0x1b9
github.com/brimsec/zq/expr.(*Shaper).orderRecordType(0xc000132360, 0xc0003bc300, 0xc0003bc300, 0xc0003bc300, 0x0, 0x0)
	/Users/phil/work/zq/expr/shaper.go:404 +0xa10
github.com/brimsec/zq/expr.(*Shaper).createShapeSpec(0xc000132360, 0xc0003bc5a0, 0xc0003bc300, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/phil/work/zq/expr/shaper.go:323 +0x1ee
github.com/brimsec/zq/expr.(*Shaper).Eval(0xc000132360, 0xc0000325f0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000032501, 0xc000128c60)
	/Users/phil/work/zq/expr/shaper.go:285 +0x337
github.com/brimsec/zq/proc/put.(*Proc).eval(0xc0000a62d0, 0xc0000325f0, 0x0, 0xc000128c60, 0xc000625d98, 0x15b87dc, 0xc000128080)
	/Users/phil/work/zq/proc/put/put.go:84 +0x127
github.com/brimsec/zq/proc/put.(*Proc).put(0xc0000a62d0, 0xc0000325f0, 0xc0000325f0, 0xc000010040, 0x0)
	/Users/phil/work/zq/proc/put/put.go:347 +0x50
github.com/brimsec/zq/proc/put.(*Proc).Pull(0xc0000a62d0, 0x0, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/proc/put/put.go:371 +0x15f
github.com/brimsec/zq/driver.(*mux).safeGet(0xc0001282a0, 0x0, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/driver/mux.go:51 +0x6c
github.com/brimsec/zq/driver.(*mux).run(0xc0001282a0)
	/Users/phil/work/zq/driver/mux.go:64 +0x32
created by github.com/brimsec/zq/driver.(*muxOutput).Pull.func1
	/Users/phil/work/zq/driver/mux.go:98 +0x5d
```

**Comment by henridf** [comment]
Support for aliases/typedefs is #2192 and is addressed in #2219.

**Comment by philrz** [comment]
Verified in `zq` commit 49c5a2f.

Repeating the original repro steps, there's no longer a panic.

```
$ echo '{"foo": "bar"}' | zq -z -I example.zs -
{_path:null (string),ts:null (time),uid:null (bstring),id:null (0=({orig_h:ip,orig_p:port=(uint16),resp_h:ip,resp_p:port}))} (=1)
```

Thanks @henridf!
